Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -182,4 +184,55 @@ Value StringPrototype::to_string(Interpreter& interpreter)
     return js_string(interpreter, string_object->primitive_string()->string());
 }
 
+enum class PadPlacement {
+    Start,
+    End,
+};
+
+static Value pad_string(Interpreter& interpreter, Object* object, PadPlacement placement)
+{
+    auto string = object->to_string().as_string()->string();
+    if (interpreter.argument(0).to_number().is_nan()
+        || interpreter.argument(0).to_number().is_undefined()
+        || interpreter.argument(0).to_number().to_i32() < 0) {
+        return js_string(interpreter, string);
+    }
+    auto max_length = static_cast<size_t>(interpreter.argument(0).to_i32());
+    if (max_length <= string.length())
+        return js_string(interpreter, string);
+
+    String fill_string = " ";
+    if (!interpreter.argument(1).is_undefined())
+        fill_string = interpreter.argument(1).to_string();
+    if (fill_string.is_empty())
+        return js_string(interpreter, string);
+
+    auto fill_length = max_length - string.length();
+
+    String filler = "";
+    while (filler.length() < fill_length)
+        filler = String::format("%s%s", filler.characters(), fill_string.characters());
```

Comment:
Let's use a `StringBuilder` here instead of repeatedly creating new `String` objects.

Reply:
That's much better... Still making myself familiar with all the cool things in `AK` :blush: